ui/store/actions.js

```diff
@@ -51,9 +52,37 @@ import * as actionConstants from './actionConstants';
 
 let background = null;
 let promisifiedBackground = null;
+const retryQueue = [];
+
+const submitRequestToBackground = async (method, args) => {
+  if (isManifestV3) {
+    if (background.connectionStream.readable) {
+      await promisifiedBackground[method](...args);
+    } else {
+      retryQueue.push({ method, args });
+    }
+  } else {
+    await promisifiedBackground[method](...args);
+  }
+};
```

Comment:
Here we need to handle an edge case when service worker dies when request is in progress.

We can check the exception returned from backend to find reason of request failure and retry request.

Reply:
I wonder if we can use the background sync for the edge cases. Sending an action can be retried when the connection is re-established. 

ref: https://developer.chrome.com/blog/background-sync/#how-to-request-a-background-sync